src/docs.js

```diff
@@ -6,38 +6,48 @@ let path = require('path');
 let recursiveReadSync = require('recursive-readdir-sync');
 
 async function documenter(options) {
-
   options = _.defaults({}, options, {
+    metadata: null,
     schemas: [],
     docsFolder: null,
+    references: [],
   });
   assert(options.schemas, 'options.schemas must be given');
   assert(options.schemas instanceof Array, 'options.schemas must be an array');
 
-  let schemas = options.schemas;
   let tarball = tar.pack();
+  let metadata = options.metadata;
+
+  if (metadata) {
+    let data = JSON.stringify(metadata, null, 2);
+    tarball.entry({name: 'metadata.json'}, data);
```

Comment:
what is metadata supposed to be?

If this some dictionary of something complex I suggest you define JSON schema and validate against it...
otherwise at least test that it has the required properties of correct type, etc...